### PR TITLE
Update compat and remove redundant fixed

### PR DESCRIPTION
**Pull request by JoeyT1994**
This PR updates the compat on `ITensorNetworks` and `Graphs` to more recent versions and removes functions in `fixes.jl` which were addressed by the PRs in https://github.com/ITensor/ITensorNetworks.jl/pull/195 and https://github.com/ITensor/ITensorNetworks.jl/pull/194

### CODE REVIEW

**Comment by ryanlevy** [comment]
Looks good besides formating, some requests/questions:

- Can you bring the workflows up to the current version in here too? (see e.g. [ITensors.jl](https://github.com/ITensor/ITensors.jl/blob/main/.github/workflows/format_check.yml), using `actions/checkout@v4` )
- Do we want to add the dependabot?
- currently using `fixes.jl` fails for Julia 1.11.0 (they have an open bug about it), any chance we can get rid of it now or overload the `add` function differently?

**Comment by JoeyT1994** [comment]
Sure I just bought up the workflows to checkout@v4 and setup-julia@v2 . I also formatted.

Not sure what to do about `fixes.jl` not working with Julia 1.11.0, do you think they will resolve that issue soon? Quite useful to have a `fixes.jl` here so we can overload `ITensorNetworks` functionality with ease -- but I'm open to other suggestions.

File: Project.toml [hunk 1]
```diff
@@ -6,18 +6,21 @@ version = "0.1.1"
 [deps]
 Dictionaries = "85a47980-9c8c-11e8-2b9f-f7ca1fa99fb4"
 Glob = "c27321d9-0574-5035-807b-f59d2c89b15c"
+GraphRecipes = "bd48cda9-67a9-57be-86fa-5b3c104eda73"
 Graphs = "86223c79-3864-5bf0-83f7-82e725a168b6"
 ITensorNetworks = "2919e153-833c-4bdc-8836-1ea460a35fc7"
 ITensors = "9136182c-28ba-11e9-034c-db9fb085ebd5"
+JuliaFormatter = "98e50ef6-434e-11e9-1051-2b60c6c9e899"
```
Comment (ryanlevy) [inline Project.toml:13]:
```suggestion
```
Let's not make that a dependency

**Comment by ryanlevy** [comment]
Wow that new formatter really screwed up some things. I think I've sorted everything out including the `fixes.jl`, let me know if everything works on your end and then I'm good with merging

**Review by ryanlevy — APPROVED** [review]

**Comment by JoeyT1994** [comment]
Thanks! Yeah no idea what that new formatter version is about. Looks good now though.

**Pull request merged**